src/smartpeak/source/core/RawDataProcessor.cpp

```diff
@@ -1844,4 +1844,206 @@ namespace SmartPeak
     rawDataHandler_IO.clearNonSharedData();
     LOGD << "END ClearData";
   }
+
+  void CalculateMDVs::process(
+    RawDataHandler& rawDataHandler_IO,
+    const std::map<std::string, std::vector<std::map<std::string, std::string>>>& params_I,
+    const Filenames& filenames
+  ) const
+  {
+    LOGD << "START CalculateMDVs";
+
+    if (params_I.count("CalculateMDVs") && params_I.at("CalculateMDVs").empty()) {
+      LOGE << "No parameters passed to CalculateMDVs.";
+      LOGD << "END CalculateMDVs";
+      return;
+    }
+
+    OpenMS::IsotopeLabelingMDVs isotopelabelingmdvs;
+    OpenMS::Param parameters = isotopelabelingmdvs.getParameters();
+    //Utilities::updateParameters(parameters, params_I.at("calculateMDVs"));
+    isotopelabelingmdvs.setParameters(parameters);
+
+    try {
+      OpenMS::FeatureMap normalized_featureMap;
+      std::vector<std::map<std::string, std::string>> CalculateMDVs_params = params_I.find("CalculateMDVs")->second;
+      
+      std::string mass_intensity_type, feature_name;
+      for(auto& param : CalculateMDVs_params)
+      {
+        if (param.find("name")->second == "mass_intensity_type")
+        {
+          mass_intensity_type = param.find("value")->second;
+        }
+        else if (param.find("name")->second == "feature_name")
+        {
+          feature_name = param.find("value")->second;
+        }
+      }
+      
+      isotopelabelingmdvs.calculateMDVs(rawDataHandler_IO.getFeatureMap(), normalized_featureMap, mass_intensity_type, feature_name);
+      rawDataHandler_IO.setFeatureMap(normalized_featureMap);
```

Comment:
Don't forget the call to `rawDataHandler_IO.updateFeatureMapHistory();` to update the Feature map history with the new feature data.

Reply:
Also in the rest of the new methods.

Reply:
Done.